From fd329bac32dbf4038d458a8f019ebbb42e8faf48 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Wed, 20 Apr 2016 16:54:08 +0200
Subject: [PATCH] Add AppStream metadata

This will ensure KiCad will be easily installable from the app stores such as
GNOME Software.

The screenshots probably need a better home. Not that it would matter
too much -- the Linux distributions do cache them with their feeds, so
the app store applications don't access them from this source directly.

Screenshots from the web [1] can't be used since they are too large for use
in the app store application. :(

[1] http://kicad-pcb.org/discover/screenshots/
---
 CMakeLists.txt                            |  7 +++++
 resources/linux/appdata/kicad.appdata.xml | 49 +++++++++++++++++++++++++++++++
 2 files changed, 56 insertions(+)
 create mode 100644 resources/linux/appdata/kicad.appdata.xml

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 456f5e3..6c5eaa1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -860,6 +860,7 @@ if( UNIX AND NOT APPLE )
     set( UNIX_MIME_FILES ${UNIX_MIME_DIR}/mime )
     set( UNIX_ICON_FILES ${UNIX_MIME_DIR}/icons )
     set( UNIX_APPLICATIONS_FILES ${UNIX_MIME_DIR}/applications )
+    set( UNIX_APPDATA_FILES resources/linux/appdata )
 
     # Install Mimelnk directory
     install( DIRECTORY ${UNIX_MIMELNK_FILES}
@@ -884,6 +885,12 @@ if( UNIX AND NOT APPLE )
         DESTINATION ${CMAKE_INSTALL_PREFIX}/share
         COMPONENT resources
         )
+
+    # Install AppStream directory (app store entry)
+    install( DIRECTORY ${UNIX_APPDATA_FILES}
+        DESTINATION ${CMAKE_INSTALL_PREFIX}/share
+        COMPONENT resources
+        )
 endif()
 
 #include( CTest )
diff --git a/resources/linux/appdata/kicad.appdata.xml b/resources/linux/appdata/kicad.appdata.xml
new file mode 100644
index 0000000..4452714
--- /dev/null
+++ b/resources/linux/appdata/kicad.appdata.xml
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright Lubomir Rintel 2016 -->
+<component type="desktop">
+  <id>kicad.desktop</id>
+  <name>KiCad</name>
+  <project_license>GPL-2.0+</project_license>
+  <metadata_license>CC0-1.0</metadata_license>
+  <summary>EDA Suite</summary>
+
+  <keywords>
+    <keyword>KiCad</keyword>
+    <keyword>EDA</keyword>
+    <keyword>PCB</keyword>
+    <keyword>Schema</keyword>
+    <keyword>PcbNew</keyword>
+    <keyword>Eeschema</keyword>
+  </keywords>
+
+  <description>
+    <p>A Cross Platform and Open Source Electronics Design Automation Suite.</p>
+    <p>The programs handle Schematic Capture, and PCB Layout with Gerber output.</p>
+  </description>
+
+  <screenshots>
+    <screenshot type="default">
+      <image width="800" height="450">https://raw.githubusercontent.com/lkundrak/kicad-source-mirror/master/resources/linux/screenshots/kicad.png</image>
+    </screenshot>
+
+    <screenshot type="default">
+      <caption>Eeschema Schematic Editor</caption>
+      <image width="800" height="450">https://raw.githubusercontent.com/lkundrak/kicad-source-mirror/master/resources/linux/screenshots/kicad-schema.png</image>
+    </screenshot>
+
+    <screenshot type="default">
+      <caption>PcbNew PCB Layout</caption>
+      <image width="800" height="450">https://raw.githubusercontent.com/lkundrak/kicad-source-mirror/master/resources/linux/screenshots/kicad-pcbnew.png</image>
+    </screenshot>
+
+    <screenshot type="default">
+      <caption>PcbNew 3D Viewer</caption>
+      <image width="800" height="450">https://raw.githubusercontent.com/lkundrak/kicad-source-mirror/master/resources/linux/screenshots/kicad-3d.png</image>
+    </screenshot>
+  </screenshots>
+
+  <url type="homepage">http://kicad-pcb.org/</url>
+  <url type="bugtracker">http://kicad-pcb.org/help/report-a-bug/</url>
+  <update_contact>kicad-developers@lists.launchpad.net</update_contact>
+  <developer_name>The KiCad Developers</developer_name>
+</component>
-- 
2.5.5

